src/zoid/modal/component.jsx

```diff
@@ -36,6 +39,9 @@ export function getModalComponent() : ModalComponent {
                 }
 
                 const { cspNonce } = props;
+                // const nonce = cspNonce || getCSPNonce() || '';
```

Comment:
Can we keep the default as `|| getCSPNonce() || ''`? This means if a merchant passes `data-csp-nonce` to the sdk (which is the recommended pattern) we will use that default value.

(Honestly, maybe we should think about making `getCSPNonce` return `string` not `?string` to avoid having to do `getCSPNonce() || ''`, but happy to punt on that for now)

Reply:
The current implementation of this is to have `cspNonc` passed in as a prop, which I think is right?   
The commented out line of was  `const nonce = cspNonce || getCSPNonce() || '';` was me messing around with trying to set a fall back so certain tests wouldn't fail.

Reply:
`getCSPNonce()` will pull the nonce from the global `data-csp-nonce` prop passed to the SDK script tag. So we should probably respect/default to that whenever it's not passed as a prop, so any merchant using that prop can be confident it will be used by any child components. Then if a prop is passed it can override that value.